keyring/backends/macOS/__init__.py

```diff
@@ -32,36 +33,54 @@ def priority(cls):
             raise RuntimeError("Security API unavailable")
         return 5
 
-    def set_password(self, service, username, password):
+    def set_password(self, service, username, password, keychain=""):
         if username is None:
             username = ''
 
+        if keychain != "":
+            if not exists(keychain):
+                raise KeyringLocked("Can't open specified keychain")
+        elif os.environ.get('KEYCHAIN_PATH') != "":
+            keychain = os.environ.get('KEYCHAIN_PATH')
+
         try:
-            api.set_generic_password(self.keychain, service, username, password)
+            api.set_generic_password(keychain, service, username, password)
         except api.KeychainDenied as e:
             raise KeyringLocked("Can't store password on keychain: " "{}".format(e))
         except api.Error as e:
             raise PasswordSetError("Can't store password on keychain: " "{}".format(e))
 
-    def get_password(self, service, username):
+    def get_password(self, service, username, keychain=""):
         if username is None:
             username = ''
 
+        if keychain != "":
+            if not exists(keychain):
+                raise KeyringLocked("Can't open specified keychain")
+        elif os.environ.get('KEYCHAIN_PATH') != "":
+            keychain = os.environ.get('KEYCHAIN_PATH')
+
```

Comment:
This code looks like a repeat of something that was done elsewhere. Better would be to implement this logic in one place.